api/cluster.go

```diff
@@ -24,3 +26,26 @@ type ReportSummary struct {
 	FailureCount int    `json:"failureCount"`
 	SuccessCount int    `json:"successCount"`
 }
+
+// UnmarshalJSON unmarshals a ClusterSummary.
+func (c *ClusterSummary) UnmarshalJSON(data []byte) error {
+	type Alias ClusterSummary
+	aux := &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(c),
+	}
```

Comment:
I'm not sure exactly what this does. [(lines)](https://github.com/jetstack/preflight/pull/89/files#diff-bc8a656be21edce0cf2a74adf23d7aeaR32-R37). :thinking:

Reply:
We cannot do this:

```
var aux ClusterSummary
if err := json.Unmarshal(data, &aux); err != nil {
		return err
}
```

That would call the `UnmarshalJSON` method we are declaring recursively with no end, which leads to stack overflow.

We need to create that artificial alias so we can unmarshal to an intermediate variable using the native unmarshal logic and then apply our custom logic.

Reply:
Ahh, that makes sense. Thanks.